src/components/table/daikin-table.ts

```diff
@@ -0,0 +1,462 @@
+   */
+  @property({ type: Boolean, reflect: true })
+  selectable: boolean = false;
+
+  /**
+   * Whether or not to enable sorting of the rows.
+   * If `true`, a button for sorting will be displayed to the right of each header cell text.
+   */
+  @property({ type: Boolean, reflect: true })
+  sortable: boolean = false;
+
+  /**
+   * An array of `id`s for the `rows` that have been checked.
+   */
+  @property({ type: Array, attribute: false })
+  selection: string[] = [];
+
+  /**
+   * Sort order of the table.
+   */
+  @property({ type: String, reflect: true })
+  order: "asc" | "desc" | null = null;
+
+  /**
+   * Specify this when you want to customize the sort function.
+   * The function must be ascending, and the return value must be `0`, `1` or `-1`.
+   */
+  @property({ attribute: false })
+  sort:
+    | {
+        [key in keyof T]?: (a: T, b: T, key: key) => number;
+      }
+    | null = null;
+
+  /**
+   * The `key` of the currently sorted column.
+   */
+  @property({ type: String, reflect: true, attribute: "sorted-key" })
+  sortedKey: keyof T | null = null;
+
+  @state()
+  private _cells: DaikinTableCell[] = [];
+
+  @state()
+  private _headerCells: DaikinTableHeaderCell[] = [];
+
+  @state()
+  private _bulkRowsCheckState: "unchecked" | "indeterminate" | "checked" =
+    "unchecked";
+
+  /**
+   * An array that performs sorting and searching based on `rows`.
+   * It does not manage checks and pagination.
+   */
+  @state()
+  private _showRows: T[] = [];
+
+  private _updateSort() {
+    this._showRows = this.rows.sort((a, b) => {
+      if (!this.sortedKey) {
+        return 0;
+      }
+
+      if (this.sort && !!this.sort[this.sortedKey]) {
+        return this.sort[this.sortedKey]?.(a, b, this.sortedKey) ?? 0;
+      } else {
+        if (a[this.sortedKey] === b[this.sortedKey]) {
+          return 0;
+        } else if (a[this.sortedKey] > b[this.sortedKey]) {
+          return 1;
+        } else {
+          return -1;
+        }
+      }
+    });
+
+    if (this.order === "desc") {
+      this._showRows.reverse();
+    }
+  }
+
+  private _updateCheck() {
+    this.selection = this.selection.filter((selectedId) =>
+      this._showRows.find(({ id }) => selectedId === id)
+    );
+
+    this._checkHeaderFunction();
+  }
+
+  private _emitChangeCheckEvent() {
+    this.dispatchEvent(
+      new CustomEvent("change-check", {
+        detail: {
+          selection: this.selection,
+        },
+      })
+    );
+  }
+
+  private _updateTable() {
+    // Reset rows
+    this._showRows = this.rows;
+
+    this._updateSort();
+    this._updateCheck();
+  }
+
+  private _updateCells() {
+    this._cells = [...this.querySelectorAll("daikin-table-cell")];
+    this._headerCells = [...this.querySelectorAll("daikin-table-header-cell")];
+  }
+
+  private _handleCheckboxRowHeaderChange(): void {
+    switch (this._bulkRowsCheckState) {
+      case "unchecked":
+      case "indeterminate":
```

Comment:
It's always good to use `event.target.checked` instead of our internal states to reduce possibility of inconsistency.

```suggestion
  private _handleCheckboxRowHeaderChange(event: Event): void {
    switch ((event.target as DaikinCheckbox).checkState) {
      case "checked":
```